src/components/Drawer/PDrawer.vue

```diff
@@ -27,40 +31,29 @@
 </script>
 
 <script lang="ts" setup>
-  import { computed, useAttrs, ref } from 'vue'
+  import { computed, useAttrs } from 'vue'
   import { useDrawer, useGlobalEventListener } from '@/compositions'
   import { PLayoutResizable } from '@/layouts'
   import { DrawerPlacement, keys } from '@/types'
   import { isKeyEvent } from '@/utilities'
 
-  const props = defineProps<{
-    open?: boolean,
+  const {
+    resizable = false,
+    placement = 'left',
+    to,
```

Comment:
nit: Then you don't need to null coalesce on the teleport element. 
```suggestion
    to = 'body',
```

Reply:
you'd think that but this doesn't work for some reason

Reply:
ohhhhhhh. it's cause the peer dependency for vue was too low

Reply:
oh, nevermind then. We should update that some time lol.

Reply:
we're on 3.4.x and props destructuring wasn't introduced till 3.5

Reply:
i'm just gonna update there here unless you have reservations?

Reply:
none, we're already using 3.5 in production so that's whats actually being used for this in apps already. Shouldn't change anything.